### ISSUE TITLE
[question] Any chance notes being rendered properly if they are markdown

### ISSUE THREAD

**Issue by kirk86** [retorquere/zotero-report-customizer#44]
Hi, I was wondering if it's possible to have the notes to render properly if they are in markdown?
 At the moment in the notes section, I can only see markdown code indicating that it's not being rendered properly.

Thanks!

**Comment by retorquere** [comment]
Zotero notes are HTML. How would I tell that the plain-text in the note is markdown?

**Comment by kirk86** [comment]
> Zotero notes are HTML

Are you sure about that cause even in that case the html tags `<br />` still are not being properly rendered when generating the report 

> How would I tell that the plain-text in the note is markdown?

I presume that would be challenging to automatically detect it but maybe you can provide an checkbox option to the user so when clicked to render markdown? Similar to how you provide option to edit different sections of the report maybe?

**Comment by retorquere** [comment]
100% sure. Right-click a note and select "source". If you enter `<br />` that gets turned into `&lt;br /&gt;` by the editor.

I'll think about such a checkbox, but it'd be a major change. I can do the editing in the customizer currently by just applying CSS. Togglable markdown is going to require a reload of the report, and that would destroy the history. I might just make it a preference rather than an UI toggle.

**Comment by kirk86** [comment]
Any viable option would be much appreciated.
Right now I only use the markdown because it allows me to have latex code in the notes, otherwise I don't have use for the notes without latex.

When I generate the report I only keep the title of the paper and my notes describing it which gives me a nice overview to further work on, but without notes being rendered properly its kind of difficult to take full advantage of the report generated.

**Comment by retorquere** [comment]
LaTeX support is not part of markdown, although some markdown processors might have it. If they do, they most likely use an actual LaTeX processor under the hood. I could in principle render markdown, but LaTeX render is exceedingly unlikely to be part of that.

If all you need is math display, the report customizer can already do mathjax.

**Comment by kirk86** [comment]
As you might already know this I don't know how things work behind the scenes.
At the moment I write my notes in markdown in zotero note editor, see below an example
```
# Summary: Statistical Mechanics of Neural Networks

| Summary | Notes |
|-----------------------|----------------------------------------------------------------------------------------------------------|
| Idea 1: | This is idea 1 <br /><br /> With some latex code $x^{2} + y^{2} = r^{2}$|
| Idea 2: | This is the second idea with additional latex $x^{2} + y^{2} + z^{2} = r^{2}$|
```

Which renders into 
![image](https://user-images.githubusercontent.com/2902390/73676195-dda2b000-46ab-11ea-9ba8-41b301bbfe5d.png)

With the nice help of `Markdown Here` plugin in Zotero.

Now, all I want is what is rendered above to be somehow included in the generated reports. How can I achieve that?

**Comment by kirk86** [comment]
@retorquere Never mind, I made a stupid mistake, the generated report actually can include the rendered markdown I just had to render it first properly using markdown here and then generate the report. If you want you can close this issue, I can close it if you want?

**Comment by retorquere** [comment]
MathJax support can help with those latex parts. But rendering markdown is likely to interfere with this.

**Comment by kirk86** [comment]
> MathJax support can help with those latex parts

Is that only for the generated report? Will I be able to see the generated result (i.e. rendered stuff) in the note editor in Zoter?

> But rendering markdown is likely to interfere with this.

Trying to answer my own question, just created a note with only text and a latex fragment and the rendering is neither working in the note editor nor in the generated report.

**Comment by retorquere** [comment]
MathJax only renders in the generated report.

For the generated report you need to either use the default mathjax markers, or reconfigure the report customizer for latex-compatible markers. See the readme.

**Comment by kirk86** [comment]
Thanks, I'll stick to what I have since it fits my needs better. I'm closing this for now so that it doesn't provide additional confusion.

**Comment by retorquere** [comment]
> @retorquere Never mind, I made a stupid mistake, the generated report actually can include the rendered markdown I just had to render it first properly using markdown here and then generate the report. If you want you can close this issue, I can close it if you want?

I had missed this, great you found a workable solution.
